transformations/unit_converter/transformation.py

```diff
@@ -0,0 +1,135 @@
+import itertools
+import json
+import os
+import random
+
+import spacy
+from num2words import num2words
+from word2number import w2n
+
+from initialize import spacy_nlp
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+
+def convert(text, X, units, plurals, converter):
+    is_word, not_zero = False, False
+    if str(X.root) in units:  # check if the unit is supported
+        q = str(X.text).replace(str(X.root), "")  # q is the number of units
+        try:
+            q = float(q)
+        except ValueError:
+            q = w2n.word_to_num(
+                q
+            )  # if it is not a numerical value, convert it and remember to convert it back after unit conversion
+            is_word = True
+        q /= converter[
+            plurals[X.root.lemma_]
+        ]  # convert to base unit (meters, kilograms)
+        new_unit = random.choice(
```

Comment:
It appears that you choose one new unit, even in cases a sentence contains both weight and length measures. Would you consider changing this or is this against your intention? 
Your example sentence "The phone is zero point zero five five yards wide and it weights 5.997 ounces." would only ever modify the weight measure.

Reply:
It chooses one new unit for each entity detected by spacy (you can see the for loop in line 69). This means that in the 4th example in test.json the transformation converts both centimeters to yards and grams to ounces.

Reply:
Apologies, I realised that I took the _output_ of your example sentence as input. 

Indeed, when I run the actual input example "The phone is five centimeters wide and it weights 170 grams." 100 times (after disabling the random seed) and compare the outputs, we get different combinations of outputs for the first and second measure. So far, so good. 
 
However, when I run the output "_The phone is zero point zero five five yards wide and it weights 5.997 ounces"_ 100 times and compare the first part of the output [12:50], all returned first parts are equal while only the second half (weight measure) differs. 
This may mean that spacy does not detect _"zero point zero five five yards"_.

Reply:
Yes, you are right. Unfortunately spacy does not detect "zero point zero five five yards" as "QUANTITY", but it should detect "0.055 yards" as "QUANTITY", that should be much common to find in written texts.

Reply:
I agree, this is highly unusual. I only brought it up as this quantity might be lost when repeatedly running the transformation on itself. But again, this might be out of scope here.